node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -211,6 +213,165 @@ fn add_vote() -> Result<(), AddVoteError<TestContext>> {
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 51.into(),
+        round_exp: 4u8,
+        endorsed: vec![],
+    };
+    assert_eq!(state.validate_lnc(&c0), Err(LncError::NaiveCitation(ALICE)));
+    // c0 claims that b0 is endorsed.
+    c0.endorsed = vec![b0];
+    // but we don't have these endorsements in the protocol state.
+    assert_eq!(
+        state.validate_lnc(&c0),
+        Err(LncError::MissingEndorsement(b0))
+    );
+    // Add endorsements for b0.
+    endorse!(state, rng, CAROL, b0);
+    assert_eq!(
+        state.validate_lnc(&c0),
+        Err(LncError::MissingEndorsement(b0))
+    );
+    endorse!(state, rng, BOB, b0);
+    // Now c0 cites ALICE's equivocations non-naively b/c b0 is endorsed.
+    assert_eq!(state.validate_lnc(&c0), Ok(()));
+
+    // 3. Equivocation cited by two honest validators in the vote's panorama – their votes need to
+    // be endorsed.
+    //
+    // Bob:      b0
+    //          / \
+    // Alice: a0   \
+    //              \
+    //        a0'    \
+    //           \   |
+    // Carol:    c0  |
+    //             \ |
+    // Dan:         d0
+
+    let weights_dan = &[Weight(3), Weight(4), Weight(5), Weight(5)];
+    let mut state = State::new_test(weights_dan, 0);
+    let a0 = add_vote!(state, rng, ALICE, 0xA; N, N, N, N)?;
+    let a0_prime = add_vote!(state, rng, ALICE, 0xA; N, N, N, N)?;
+    let b0 = add_vote!(state, rng, BOB, 48, 4u8, 0xB; a0, N, N, N)?;
+    let c0 = add_vote!(state, rng, CAROL, 48, 4u8, 0xB; a0_prime, N, N, N)?;
+    // d0 violates LNC b/c it naively cites Alice's equivocation.
+    let mut d0 = WireVote {
+        panorama: panorama!(F, b0, c0),
+        creator: DAN,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 51.into(),
+        round_exp: 4u8,
+        endorsed: vec![],
+    };
+    // None of the votes is marked as being endorsed – violates LNC.
+    assert_eq!(state.validate_lnc(&d0), Err(LncError::NaiveCitation(ALICE)));
+    d0.endorsed = vec![c0];
+    endorse!(state, rng, CAROL, c0);
+    // One endorsement isn't enough.
+    assert_eq!(
+        state.validate_lnc(&d0),
+        Err(LncError::MissingEndorsement(c0))
+    );
+    endorse!(state, rng, BOB, c0);
+    endorse!(state, rng, DAN, c0);
+    // Now d0 cites non-naively b/c c0 is endorsed.
+    assert_eq!(state.validate_lnc(&d0), Ok(()));
+
+    // 4. Multiple equivocators and indirect equivocations.
+    // Votes are seen as endorsed by `state` – does not violate LNC.
+    //
+    // Alice   a0<---------+
+    //                     |
+    //         a0'<--+     |
+    //               +     |
+    // Bob          b0<-----------+
+    //               +     |      |
+    // Carol   c0<---+     |      |
+    //                     |      |
+    //         c0'<--------+      |
+    //                     |      |
+    // Dan                 d0<----+
+    //                            +
```

Comment:
```suggestion
    //                            |
```
(Also above and below `b0`.)